core/src/main/java/com/nisovin/magicspells/spells/PlayerMenuSpell.java

```diff
@@ -263,7 +293,7 @@ public void onItemClick(InventoryClickEvent event) {
 		if (variableTarget != null && !variableTarget.isEmpty() && MagicSpells.getVariableManager().getVariable(variableTarget) != null) {
 			MagicSpells.getVariableManager().set(variableTarget, player, target.getName());
 		}
-		if (stayOpen) open(player, data);
+		if (stayOpen) open(player, data, 0);
```

Comment:
This would reset page to the first page, even if you were on another page. I suggest modifying the `spellData` map to use another record that includes current page, such as `record MenuData(SpellData, int page)`, to preserve that information.